src/libraries/Client/Microsoft.Agents.CopilotStudio.Client/UserAgentHelper.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http.Headers;
+using System.Reflection;
+
+namespace Microsoft.Agents.CopilotStudio.Client
+{
+    internal static class UserAgentHelper
+    {
+        private static object _initLock = new object();
+        public static string ClientFileVersion { get; private set; }
+
+        public static string ClientName { get; private set; } = "CopilotStudioClient";
+
+        public static ProductInfoHeaderValue UserAgentHeader { get; private set; }
+
+        static UserAgentHelper()
+        {
+            lock (_initLock)
+            {
+                if (string.IsNullOrEmpty(ClientFileVersion))
+                {
+                    ClientFileVersion = "unknown";
+                    // Get the version of the assembly
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
+#pragma warning disable CS8601 // Possible null reference assignment.
+                    try
+                    {
+                        ClientFileVersion = typeof(CopilotClient).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version ?? FileVersionInfo.GetVersionInfo(typeof(CopilotClient).Assembly.Location).FileVersion;

```

Comment:
you can use `ThisAssembly` to avoid using reflection

Reply:
I do not want to pull in the Nerdbank Lib into the release package.

Reply:
you dont need to, the nbgv is used only at build time, and injects the ThisAssembly class with the version stamped on it.

Reply:
what's the concern of using `ThisAssembly` ?

Reply:
In this case its fine, 
for other things, I'm trying to target an assembly other then the current executing assembly

Reply:
I dont see the ThisVersion, do you want me to push an update to this branch?